src/main/java/org/folio/service/orders/flows/update/open/OpenCompositeOrderFlowValidator.java

```diff
@@ -92,12 +103,49 @@ public Future<Void> checkLocationsAndPiecesConsistency(List<CompositePoLine> poL
       });
   }
 
-  private void validateMaterialTypes(CompositePurchaseOrder purchaseOrder){
+  private void validateMaterialTypes(CompositePurchaseOrder purchaseOrder) {
     if (purchaseOrder.getWorkflowStatus() != PENDING) {
       List<Error> errors = compositePoLineValidationService.checkMaterialsAvailability(purchaseOrder.getCompositePoLines());
       if (!errors.isEmpty()) {
         throw new HttpException(422, errors.get(0));
       }
     }
   }
+
+  public Future<Void> checkFundLocationRestrictions(List<CompositePoLine> poLines, RequestContext requestContext) {
+    logger.debug("checkFundLocationRestrictions start");
+    List<Future<Void>> checkFunds = poLines
+      .stream()
+      .filter(poLine -> CollectionUtils.isNotEmpty(poLine.getLocations()))
+      .map(poLine -> {
+        List<String> fundIdList = poLine.getFundDistribution().stream()
+          .map(FundDistribution::getFundId)
+          .toList();
+        Future<List<Fund>> future = fundService.getFunds(fundIdList, requestContext);
+        return future.compose(funds -> validateLocationRestrictions(poLine, funds));
```

Comment:
would be more readable
```suggestion
        return fundService.getFunds(fundIdList, requestContext)
          .compose(funds -> validateLocationRestrictions(poLine, funds));
```

Reply:
done